### PR TITLE
Enhancement: add additional validations for name title prefix field

### PR DESCRIPTION
**Pull request by glaubersilva**
Resolves [GIVE-1289]

## Description

This PR implements a defensive approach to always check if the name title prefix is enabled on the form before accepting its values on the form submission.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [ ] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [ ] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

[GIVE-1286]: https://stellarwp.atlassian.net/browse/GIVE-1289?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[GIVE-1289]: https://stellarwp.atlassian.net/browse/GIVE-1289?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: includes/process-donation.php [hunk 1]
```diff
@@ -1643,16 +1644,28 @@ function give_validate_required_form_fields( $form_id ) {
...
+	$formId = absint( $post_data['give-form-id'] );
 
-	if ( ! $is_alpha_first_name || ( ! empty( $post_data['give_last'] ) && ! $is_alpha_last_name ) ) {
-		give_set_error( 'invalid_name', esc_html__( 'The First Name and Last Name fields cannot contain an email address or numbers.', 'give' ) );
-	}
+    if (!give_is_name_title_prefix_enabled($formId) && isset($post_data['give_title'])) {
+          give_set_error( 'disabled_name_title', esc_html__( 'The name title prefix field is not enabled.', 'give' ) );
+    }
+
+    if (give_is_name_title_prefix_enabled($formId) && !in_array($post_data['give_title'], array_keys(give_get_name_title_prefixes($formId)))) {
```
Comment (jonwaldstein) [inline includes/process-donation.php:1660]: We'll want to first check if title isset first or else will always fail

```suggestion
    if (give_is_name_title_prefix_enabled($formId) && isset($post_data['give_title']) && !in_array($post_data['give_title'], array_keys(give_get_name_title_prefixes($formId)))) {
```

File: includes/process-donation.php [hunk 2]
```diff
@@ -1643,16 +1644,28 @@ function give_validate_required_form_fields( $form_id ) {
...
+      give_set_error( 'invalid_name_title', esc_html__( 'The name title prefix field is not valid.', 'give' ) );
+    }
+
+    $is_alpha_first_name = ( ! is_email( $post_data['give_first'] ) && ! preg_match( '~[0-9]~', $post_data['give_first'] ) );
+    $is_alpha_last_name  = ( ! is_email( $post_data['give_last'] ) && ! preg_match( '~[0-9]~', $post_data['give_last'] ) );
+    $is_alpha_title = ( ! is_email( $post_data['give_title'] ) && ! preg_match( '~[0-9]~', $post_data['give_title'] ) );
+
+    if (!$is_alpha_title || ! $is_alpha_first_name || ( ! empty( $post_data['give_last'] ) && ! $is_alpha_last_name) ) {
```
Comment (jonwaldstein) [inline includes/process-donation.php:1668]: We'll need to first check if title not empty first or else will always fail

```suggestion
    if ($is_alpha_first_name || ( ! empty( $post_data['give_last'] ) && ! $is_alpha_last_name) || ( ! empty( $post_data['give_title'] ) && ! $is_alpha_title) ) {
```

**Review by jonwaldstein — CHANGES_REQUESTED** [review]

File: includes/process-donation.php (hunk 1 above)
Comment (glaubersilva) [reply includes/process-donation.php]: Resolved in d96d7605db1f2706a357d259a2d4f2bc61e6ab90

File: includes/process-donation.php (hunk 2 above)
Comment (glaubersilva) [reply includes/process-donation.php]: Resolved in d96d7605db1f2706a357d259a2d4f2bc61e6ab90

**Review by jonwaldstein — APPROVED** [review]
Nice job!

**Review by rickalday — APPROVED** [review]
Passed manual QA tests.

**Pull request merged**